src/raster/array.jl

```diff
@@ -21,9 +21,10 @@ us to be able to index into it like we would an array.
 Constructing a RasterDataset will error if the raster bands do not have all the
 same size and a common element data type.
 """
-struct RasterDataset{T,DS<:AbstractDataset} <: AbstractDiskArray{T,3}
+struct RasterDataset{T,DS<:AbstractDataset,C} <: AbstractDiskArray{T,3}
```

Comment:
```suggestion
struct RasterDataset{T,DS<:AbstractDataset,C<:DiskArrays.GridChunks} <: AbstractDiskArray{T,3}
```

We can put the type on the parameter

It's a wrapper for a tuple of RegularChunks AbstracVectors, that lazily generate the ranges for each axis of the chunking